torch/onnx/_internal/exporter.py

```diff
@@ -75,14 +82,179 @@ class ONNXFakeContext:
     """List of paths of files that contain the model `state_dict`"""
 
 
+class OnnxRegistry:
+    """Registry for ONNX functions.
+
+    The registry maintains a mapping from qualified names to symbolic functions under a
+    fixed opset version. It supports registering custom onnx-script functions and for
+    dispatcher to dispatch calls to the appropriate function.
+
+    Attributes:
+        _registry: The registry maps OpNameto a list of SymbolicFunctions. It is important
+            not to directly modify this variable. Instead, access to it should be done through
+            the public methods: register_custom_op, get_functions, and is_registered_op.
+
+    """
+
+    def __init__(self) -> None:
+        """Initializes the registry"""
+        self._registry: Dict[
+            registration.OpName, List[registration.SymbolicFunction]
+        ] = defaultdict(list)
+        # FIXME: Avoid importing onnxscript into torch
+        from onnxscript.function_libs.torch_lib import (  # type: ignore[import]  # noqa: F401
+            ops,  # TODO(titaiwang): get rid of this import
+            registration,
+        )
+
+        # opset_version is unused for now, since torchlib only supports opset18.
+        self._opset_version = _DEFAULT_OPSET_VERSION
```

Comment:
nit: can we get this from onnxscript torchlib?

Reply:
We don’t have that info but we can if needed

Reply:
+1
Not having to modify the exporter (different repo) each time ONNX Script changes the default opset version is indeed a good feature. ONNX Script and exporter should be able to evolve independently.

Reply:
It's a good idea to have the version from torchlib, but it doesn't officially have it now. I think we can keep this as it's not public, and wait until we are working on versioning, then we can figure out a more thorough way to connect converter registry and torchlib registry.

Reply:
I will put a TODO